### PR TITLE
cmake: update and improve configuration with templates

### LINKED ISSUE #971: set cmake_policy in mxe-conf.cmake
Currently `cmake_policy` is set in `src/cmake/test/CMakeLists.txt` which is included by some cmake-based builds. I think, it worth moving them to `mxe-conf.cmake` as they are applicable to all cmake-based builds related to MXE.

### PR DESCRIPTION
**Pull request by tonytheodore**
- FindOpenSSL module was needed before cmake was included in MXE: https://github.com/mxe/mxe/commit/6d9755682567968d5f7bdbd10bf436fbd157801a
 - move cmake configuration from mxe-conf to cmake-conf
 - replace `echo` with templates for readability and maintenance
 - allow packages to set other dep files
 - set CMAKE_POLICY_DEFAULT_CMPNNNN in wrapper since `cmake_minimum_required` or `cmake_policy` can't be set in toolchain (closes #971)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -507,6 +507,7 @@ $(1): $(PREFIX)/$(3)/installed/$(1)
 $(PREFIX)/$(3)/installed/$(1): $(PKG_MAKEFILES) \
                           $(PKG_PATCHES) \
                           $(PKG_TESTFILES) \
+                          $($(PKG)_FILE_DEPS) \
```
Comment (starius) [inline Makefile]: In other parts of this definition `$(1)` is used instead of `$(PKG)`.
Can `$(PKG)` be used for all of them to support consistency and readability?

Comment (tonytheodore) [inline Makefile]: Yes, generally references to `$(1)` and `$(3)` can be replaced with `$(PKG)` and `$(TARGET)`. I think that should be a separate PR though.

**Comment by starius** [comment]
> +155 −1,004

Wow! Great!

I am building this PR applied to latest master (b76b559f5ea40dacbb887405d6a75a4bbfc9d203). The logs will be available in few days.

Can you rebase it, please?

File: Makefile (hunk 1 above)
Comment (starius) [inline Makefile]: Sure, it should be a separate PR. I would use `$(1)` here to maintain consistency meanwhile.

**Comment by tonytheodore** [comment]
Rebased with `$(1)` instead of `$(PKG)` and [https merge](https://github.com/mxe/mxe/pull/1660/commits/24d37efa5dae04fbe143c33b8c6710a8606c5451#diff-110007fd93c2d2f924fa304c7cdad376R30)

**Comment by starius** [comment]
Successfully tested for all packages on all 4 targets.
Thanks!

**Comment by starius** [comment]
For some reason the Travis build had many downloading errors, including cmake.
Everything downloads well locally.
